### ISSUE TITLE
Error on https://neopets.com/games/play_flash.phtml

### ISSUE THREAD

**Issue by Hayley-l-williams** [ruffle-rs/ruffle#12362]
# Error Info
Error name: Error
Error message: panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_buffer
    Not enough memory left

', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.2/src/backend/direct.rs:3019:5
Error stack:
```
Error: panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_buffer
    Not enough memory left

', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.2/src/backend/direct.rs:3019:5
    at U.n.wbg.__wbg_new_d258248ed531ff54 (https://unpkg.com/@ruffle-rs/ruffle@0.1
<img width="623" alt="Screenshot 2023-07-27 at 4 48 05 PM" src="https://github.com/ruffle-rs/ruffle/assets/102319049/3ae7ca2e-1ac5-4740-99f3-572e47371eda">
.0-nightly.2023.7.12/core.ruffle.9ae8e4e1f9339ad84332.js:1:51428)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h1fb1bf8fc190c8e7 (wasm://wasm/02e404ce:wasm-function[9041]:0x89b6d4)
    at std::panicking::rust_panic_with_hook::h85b7f6628c291e12 (wasm://wasm/02e404ce:wasm-function[5978]:0x7a4535)
    at core::panicking::panic_fmt::hf5c4cd929d4aaa9e (wasm://wasm/02e404ce:wasm-function[8886]:0x896ef4)
    at core::ops::function::Fn::call::heaa64b0a8e603a24 (wasm://wasm/02e404ce:wasm-function[6269]:0x7c177e)
    at wgpu::backend::direct::ErrorSinkRaw::handle_error::had77c144e7bf2239 (wasm://wasm/02e404ce:wasm-function[4692]:0x73990c)
    at <T as wgpu::context::DynContext>::device_create_buffer::h5d83ac1926422977 (wasm://wasm/02e404ce:wasm-function[6860]:0x7f46d7)
    at <wgpu::Device as wgpu::util::device::DeviceExt>::create_buffer_init::h246f45328f543134 (wasm://wasm/02e404ce:wasm-function[1514]:0x45cb69)
    at core::cell::once::OnceCell<T>::get_or_try_init::hf846a7f58b0ec44b (wasm://wasm/02e404ce:wasm-function[3709]:0x6950ba)
    at ruffle_render_wgpu::surface::Surface::draw_commands_and_copy_to::h58d01cffd3140c61 (wasm://wasm/02e404ce:wasm-function[1343]:0x413930)
```

# Player Info
Allows script access: true
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (Apple, Apple M1 Max, OpenGL 4.1)"
Adapter Device Type: IntegratedGpu
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS)
Available features: Features(CLEAR_TEXTURE)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: https://neopets.com/games/play_flash.phtml?va=&game_id=212&nc_referer=&age=1&hiscore=&sp=0&questionSet=&r=474509&&width=560&height=440&quality=high&inpage=1
SWF URL: https://images.neopets.com/games/gaming_system/np6_loader_v2_48.swf

# Browser Info
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
Platform: MacIntel
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-07-12
Channel: nightly
Built: 2023-07-12T00:26:29.523Z
Commit: f0607d6a1fc94c5c56230ab06ff612d883ebf14b
Is extension: false

# Metadata
width: 1000
height: 1000
frameRate: 48
numFrames: 21
swfVersion: 6
backgroundColor: #000000
isActionScript3: false
uncompressedLength: 8635

**Comment by n0samu** [comment]
Hi, can you tell us what led up to this error? Did it happen at a specific point in the game?

**Comment by Hayley-l-williams** [comment]
Yes, it happened when I completed the game ("Game over"). I switched to another browser tab for a minute and then when I returned, the error happened.

**Comment by n0samu** [comment]
Alright, thanks for letting us know! I think your browser probably noticed you hadn't been using the tab for a while, so it freed up the memory it was using, including the data Ruffle needed to keep running. Ruffle isn't able to recover from that type of situation yet, so it crashed.
